strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/PypiArtifactCoordinates.java

```diff
@@ -201,13 +263,38 @@ public void setPlatform(String platform)
     @Override
     public String toPath()
     {
-        // if optional BUILD coordinate is empty, don't include it in the reconstruction
+        String fileName = SOURCE_EXTENSION.equals(getPackaging()) ? buildSourcePackageFileName()
+                                                                  : buildWheelPackageFileName();
+
+        return String.format("%s/%s/%s", getId(), getVersion(), fileName);
+    }
+
+    private String buildSourcePackageFileName()
+    {
+        return String.format("%s-%s.%s", getId(), getVersion(), getPackaging());
+    }
+
+    private String buildWheelPackageFileName() {
+        String path;
+
         if (StringUtils.isBlank(getBuild()))
         {
-            return String.format("%s-%s-%s-%s-%s", getId(), getVersion(), getLanguageImplementationVersion(), getAbi(), getPlatform()) + ".whl";
+            path = String.format("%s-%s-%s-%s-%s.%s", getId(), getVersion(), getLanguageImplementationVersion(),
```

Comment:
Cosmetic: With multi-line parameters, we generally have one per line. Would you mind changing this? Thanks!

Reply:
fixed